### PR TITLE
refactor: Natspec and Variables

### PR DESCRIPTION
**Pull request by deluca-mike**
# Description

# Integrations Checklist

- [ ] Have any function signatures changed? If yes, outline below.
- [ ] Have any features changed or been added? If yes, outline below.
- [ ] Have any events changed or been added? If yes, outline below.
- [ ] Has all documentation been updated?

# Changelog

### CODE REVIEW

**Review by JGcarv — APPROVED** [review]

**Review by satyamakgec — APPROVED** [review]

File: contracts/LoanPrimitive.sol [hunk 1]
```diff
@@ -185,41 +191,45 @@ contract LoanPrimitive {
...
     /*******************************/
     /*** Internal Pure Functions ***/
     /*******************************/
 
-    /**
-     *  @dev Returns the fee by applying an annualized fee rate over an interval of time.
-     */
-    function _getFee(uint256 amount, uint256 feeRate, uint256 interval) internal pure virtual returns (uint256) {
-        return amount * _getPeriodicFeeRate(feeRate, interval) / uint256(10_000 * 100);
+    /// @dev Returns a fee by applying an annualized and scaled fee rate, to an amount, over an interval of time.
```
Comment (lucas-manuel) [inline contracts/LoanPrimitive.sol:213]: If Natspec in internal functions can be consumed by docgen, we should have params for all of these functions. If not, this is fine as is.

File: contracts/LoanPrimitive.sol [hunk 2]
```diff
@@ -185,41 +191,45 @@ contract LoanPrimitive {
...
+    /// @dev Returns principal and interest fee portions of a payment instalment, given generic, stateless loan parameters.
+    function _getInstallment(uint256 principal_, uint256 endingPrincipal_, uint256 interestRate_, uint256 paymentInterval_, uint256 totalPayments_)
+        internal pure virtual returns (uint256 principalAmount_, uint256 interestAmount_)
     {
-        uint256 periodicRate = _getPeriodicFeeRate(interestRate, paymentInterval);
-        uint256 raisedRate   = _scaledExponent(uint256(10_000 * 100) + periodicRate, totalPayments, uint256(10_000 * 100));
+        /**
+         *           |                                                                |   |                 RATE                |
+         *  AMOUNT = | ( PRINCIPAL * (1 + RATE) ^ TOTAL_PAYMENTS ) - ENDING_PRINCIPAL | * | ----------------------------------- |
+         *           |                                                                |   | ( (1 + RATE) ^ TOTAL_PAYMENTS ) - 1 |
```
Comment (lucas-manuel) [inline contracts/LoanPrimitive.sol:225]:
```
         *            /                                                               \     /                 RATE                \
         *  AMOUNT = | ( PRINCIPAL * (1 + RATE) ^ TOTAL_PAYMENTS ) - ENDING_PRINCIPAL  | * | ------------------------------------  |
         *            \                                                               /     \ ( (1 + RATE) ^ TOTAL_PAYMENTS ) - 1 /
```

Comment (lucas-manuel) [reply contracts/LoanPrimitive.sol:225]: We could also do something like what balancer does:
```
    /**********************************************************************************************
    // calcInGivenOut                                                                            //
    // aI = tokenAmountIn                                                                        //
    // bO = tokenBalanceOut               /  /     bO      \    (wO / wI)      \                 //
    // bI = tokenBalanceIn          bI * |  | ------------  | ^            - 1  |                //
    // aO = tokenAmountOut    aI =        \  \ ( bO - aO ) /                   /                 //
    // wI = tokenWeightIn           --------------------------------------------                 //
    // wO = tokenWeightOut                          ( 1 - sF )                                   //
    // sF = swapFee                                                                              //
    **********************************************************************************************/
```

Comment (lucas-manuel) [reply contracts/LoanPrimitive.sol:225]: I do like that box formatting

**Review by lucas-manuel — COMMENTED** [review]
Depending on the outcome of the comments we might have to make another PR

**Pull request merged**